generated/nifake/nifake/unit_tests/test_library_interpreter.py

```diff
@@ -410,6 +409,49 @@ def test_read_with_warning(self):
             assert test_error_desc in str(w[0].message)
             assert f'Warning {test_error_code} occurred.' in str(w[0].message)
 
+    def test_library_singleton_get_returns_same_instance(self):
+        # Test that if library singleton get is called multiple times it returns the same instance
+        interpreter = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        assert interpreter._library is nifake._library_interpreter.LibraryInterpreter('windows-1251')._library
+
+    def test_set_runtime_environment(self):
+        # Test that SetRuntimeEnvironment is called when library interpreter is initialized
+        self.patched_library_singleton_get.stop()
+        self.patched_library_singleton_lib = patch('nifake._library.Library', return_value=self.patched_library)
+        self.patched_library.niFake_SetRuntimeEnvironment.side_effect = self.side_effects_helper.niFake_SetRuntimeEnvironment
+        self.patched_library_singleton_lib.start()
+        self.get_initialized_library_interpreter()
+        self.patched_library.niFake_SetRuntimeEnvironment.assert_called_once()
+        self.patched_library_singleton_lib.stop()
+        self.patched_library_singleton_get.start()
+
+    def test_set_runtime_environment_not_defined(self):
```

Comment:
for consistent terminology:

`test_set_runtime_environment_not_present_in_library_interpreter`